app/helpers/templates_helper.rb

```diff
@@ -28,6 +28,20 @@ def infobox_functions_and_macros(docs_section)
           :rel => 'external')
   end
 
+  def safemode_methods
+    Rails.cache.fetch("safemode_methods", :expires_in => 2.hours) do
```

Comment:
No point in using the Rails cache if you intend to create a per-process cache, use a class variable instead.